src/flb_http_client.c

```diff
@@ -536,6 +536,8 @@ static int add_host_and_content_length(struct flb_http_client *c)
     else {
         tmp = flb_sds_printf(&host, "%s:%i", out_host, out_port);
     }
+    host = tmp;
```

Comment:
this fix is not right, because the checks in line 542 where it validates ```tmp``` is because it wants to make sure tmp was properly allocated previously: 

- https://github.com/fluent/fluent-bit/blob/e21c1341cd0d87100f17c857ad8bd195e3c96719/src/flb_http_client.c#L542
- https://github.com/fluent/fluent-bit/pull/2331/files#diff-0ab7b0ab4ea13073eb72924f38a73985R537